### ISSUE TITLE
Pre-release tags are not working

### ISSUE THREAD

**Issue by anudeepsamaiya** [anothrNick/github-tag-action#231]
Whenever I raise a PR against develop(repo's default) branch, only patch part is getting bumped but I want to add/bump pre-release tags only while PR is open and new commits to the PR are pushed. Patch part should only get bumped once merged.

My configuration

```
name: CI

# Enable Buildkit and let compose use it to speed up image building
env:
  DOCKER_BUILDKIT: 1
  COMPOSE_DOCKER_CLI_BUILD: 1

on:
  pull_request:
    branches: [ "master", "main", "develop" ]
    paths-ignore: [ "docs/**" ]

  push:
    branches: [ "master", "main", "develop" ]
    paths-ignore: [ "docs/**" ]

concurrency:
  group: ${{ github.head_ref || github.run_id }}
  cancel-in-progress: true

jobs:
  Release:
      - name: Bump version and push tag
        id: tag_version
        uses: anothrNick/github-tag-action@1.55.0
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          DEFAULT_BUMP: patch
          WITH_V: true
```

**Comment by sbe-arg** [comment]
Unsure I understand what you are trying to achieve But this actions to create temporary pre-releases as well as releases in the same job if you want a pre-release you must have a job that defines when and how a pre-release is created/updated.

Cheers

**Comment by anudeepsamaiya** [comment]
> Unsure I understand what you are trying to achieve

What I am trying to achieve is that 
- any commits pushed to an open PR would create a pre-release tag v0.0.x-beta.1
- once merged a release tag v0.0.x would be created.
- 
> But this action do not create temporary pre-releases as well as releases in the same job if you want a pre-release you must have a job that defines when and how a pre-release is created/updated.

@sbe-arg Can you help me with an example of what options to use in each job?

Do you think the following options are used correctly, because I am getting two tags added now.

```
name: CI

# Enable Buildkit and let compose use it to speed up image building
env:
  DOCKER_BUILDKIT: 1
  COMPOSE_DOCKER_CLI_BUILD: 1

on:
  pull_request:
    branches: [ "master", "main", "develop" ]
    paths-ignore: [ "docs/**" ]

  push:
    branches: [ "master", "main", "develop" ]
    paths-ignore: [ "docs/**" ]

concurrency:
  group: ${{ github.head_ref || github.run_id }}
  cancel-in-progress: true

jobs:
  Pre-Release:
      - name: Bump version and push tag
        id: tag_version
        uses: anothrNick/github-tag-action@1.55.0
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          DEFAULT_BUMP: patch
          WITH_V: true
          PRERELEASE: true
          
  Release:
      - name: Bump version and push tag
        id: tag_version
        uses: anothrNick/github-tag-action@1.55.0
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
          DEFAULT_BUMP: patch
          WITH_V: true
          RELEASE_BRANCHES: develop
```

**Comment by sbe-arg** [comment]
You need to specify when the jobs need to run at the job level

For example when a pr is merged in the release job:
```
jobs:
  ...
    if: github.event_name == 'pull_request' && github.event.action == 'closed'
```

For example when a push is sent to a dev branch:
```
jobs:
  ...
    if: github.event_name != 'pull_request'
```
In the on push: I would personally filter this differently but it all depends on your development team strategy so I can be opinionated about it.

More information here: (do your research based on your needs)
https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows
